src/main/java/practice/StreamPractice.java

```diff
@@ -48,20 +69,39 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * Example: select people of working age
      * (from 18 y.o. and to 60 y.o. for men and to 55 y.o. for women inclusively).
      */
+
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
-    }
+        class FilterNecessaryPeople implements Predicate<Person> {
+            @Override
+            public boolean test(Person person) {
+                return (person.getAge() >= fromAge && person.getSex() == Person.Sex.MAN
+                        && person.getAge() <= maleToAge)
+                        || (person.getAge() >= fromAge && person.getSex() == Person.Sex.WOMAN
+                        && person.getAge() <= femaleToAge);
+            }
+        }
 
+        FilterNecessaryPeople filterNecessaryPeople = new FilterNecessaryPeople();
+        return peopleList.stream()
+                .filter(filterNecessaryPeople)
+                .collect(Collectors.toList());
```

Comment:
I don't think we need to create local class here. We can make it simpler
```suggestion
        Predicate<Person> personPredicate = person -> person.getAge() >= fromAge
                && ((person.getSex() == Person.Sex.WOMAN && person.getAge() <= femaleToAge)
                || person.getSex() == Person.Sex.MAN && person.getAge() <= maleToAge);

        FilterNecessaryPeople filterNecessaryPeople = new FilterNecessaryPeople();
        return peopleList.stream()
                .filter(personPredicate)
                .collect(Collectors.toList());
```

Reply:
done